clik/external/hal_cpu/source/hal_main.cpp

```diff
@@ -62,16 +62,23 @@ class cpu_hal_platform : public hal::hal_t {
     const uint64_t local_ram_size = 8 << 20;
     hal_device_info.type = hal::hal_device_type_riscv;
     hal_device_info.word_size = sizeof(uintptr_t) * 8;
-    hal_device_info.target_name = "CPU";
+    hal_device_info.target_name = "ock cpu";
     hal_device_info.global_memory_avail =
         global_ram_size - global_mem_max_over_allocation;
     hal_device_info.shared_local_memory_size = local_ram_size;
     hal_device_info.should_link = true;
     hal_device_info.should_vectorize = false;
-    hal_device_info.preferred_vector_width = 0;
+    // TODO: This is slightly arbitrary and based on the "host" target
+    hal_device_info.preferred_vector_width = 128 / (8 * sizeof(uint8_t));
```

Comment:
`sizeof(uint8_t)` cannot ever be anything other than 1. It's not clear to me what you're trying to do here.

Reply:
you are right, but it does reflect what the host target does. The comment there says   // default to 128 bit (16 bytes), which I could add but I'm not sure if that is more help.

Reply:
Oh, I hadn't even realised that this `8` was again due to bits vs bytes. That comment would have helped me, thanks.